backend/canisters/user/impl/src/lifecycle/post_upgrade.rs

```diff
@@ -49,3 +53,24 @@ fn mark_user_canister_empty() {
         );
     })
 }
+
+fn retry_token_swaps(state: &mut RuntimeState) {
```

Comment:
The swap could still be in progress. I think you can call is_complete or something like that

Reply:
Oh yes. Thanks for checking. I had in fact meant to look at the swap creation time and only do this if > 30 mins since creation.